### PR TITLE
OOP-school-librabry: create class Nameable

### PR DESCRIPTION
**Pull request by jmoseka**
In this PR, the following requirement is implemented 

#### Interface
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

In this PR, the following requirement is implemented 

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -18,4 +23,17 @@ def can_use_services?
...
+  public
+
+  def correct_name
+    @name
+  end
 end
+
+person = Person.new(22, 'maximilianus')
+p person.correct_name
+capitalized_person = capitalize_decorator.new(person)
```
Comment (rloterh) [inline person.rb:36]: OPTIONAL
- [ ] To test your app correctly change `capitalize_decorator` to CapitalizeDecorator

File: person.rb [hunk 2]
```diff
@@ -18,4 +23,17 @@ def can_use_services?
...
+p capitalized_person.correct_name
+capitalized_trimmed_person = trimmer_decorator.new(capitalized_person)
```
Comment (rloterh) [inline person.rb:38]: OPTIONAL
- [ ] Similarly, change `trimmer_decorator` to `TrimmerDecorator`

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Mila,  

Your project is complete! Great job for nailing it right on the first attempt 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Highlights 👍
- Composition properly implemented

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_
<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (jmoseka) [reply person.rb:36]: @rloterh It seems linters does not allow that kind of naming 
Capture

File: person.rb (hunk 2 above)
Comment (jmoseka) [reply person.rb:38]: @rloterh It gave errors that's why I opted to change 

Capture

**Comment by jmoseka** [comment]
@rloterh Thank you for the review

File: person.rb (hunk 2 above)
Comment (rloterh) [reply person.rb:38]: It's okay if it's working other way round for you! 😄

**Pull request merged**